### PR TITLE
🌱 Add e2e tests for workload cluster with Kubernetes from ci/latest

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR lays the ground for testing workload clusters with Kubernetes from ci/latest

**Which issue(s) this PR fixes**:
Rif https://github.com/kubernetes-sigs/cluster-api/issues/3569

### CODE REVIEW

**Comment by fabriziopandini** [comment]
/hold
While testing things on prow

**Comment by fabriziopandini** [comment]
/hold cancel

@vincepri @srm09 this should be ready to go and it could unblock https://github.com/kubernetes-sigs/cluster-api/issues/4043 and similar works.

Please note that this PR enables testing with Kubernetes ci/latest, but currently, there is no E2E _periodic_ job doing so.

The `pull-cluster-api-e2e-k8s-latest-main` _presumbit_ job could be used to test this PR and as a template for developing such jobs.

**Review by vincepri — APPROVED** [review]
/approve
/lgtm
/milestone v0.4.0

**Pull request merged**
